cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/runtime/batch/AppWithPartitionedFileSet.java

```diff
@@ -137,6 +138,9 @@ public void initialize(MapReduceTaskContext<byte[], Put> context) throws Excepti
       Preconditions.checkArgument(inputContext instanceof PartitionedFileSetInputContext);
       PartitionedFileSetInputContext pfsInputcontext = (PartitionedFileSetInputContext) inputContext;
       Preconditions.checkNotNull(pfsInputcontext.getInputPartitionKey());
+      Preconditions.checkArgument(
```

Comment:
To test this properly, you should actually use CombineFileInputFormat and expect multiple partition keys.